src/cmakeProject.ts

```diff
@@ -2162,7 +2162,10 @@ export class CMakeProject {
 
         const buildOnLaunch = this.workspaceContext.config.buildBeforeRun;
         if (buildOnLaunch || isReconfigurationNeeded) {
-            const buildResult = await this.build([chosen.name]);
+            const defaultBuildTargets = await this.getDefaultBuildTargets();
+            const allTargetsToBuild = new Set(defaultBuildTargets);
+            allTargetsToBuild.add(chosen.name);
```

Comment:
Check if the build target is ALL, we shouldn't add the launch targets to ALL.

Reply:
No, you should check if it's `this.allTargetName`.  If that's the case, then you don't need to add the debug target to the list.

Reply:
Done.

Reply:
use `this.allTargetName` instead of `all`

Reply:
Done.